From ee8e9c84c2447d295711acdddf8ae2398a05dd6f Mon Sep 17 00:00:00 2001
From: Ahmad Abbas <ahmada@marvell.com>
Date: Thu, 11 Feb 2016 20:53:22 +0200
Subject: [PATCH 0914/1240] menuconfig: menuconfig clean up - IO Support

moved IO support configuration options under new category Machine options.

Change-Id: I89accc32a117306fb83caf5c2f490f445c098b24
Signed-off-by: Ahmad Abbas <ahmada@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27461
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Hanna Hawa <hannah@marvell.com>
---
 arch/arm/cpu/armv8/armada3700/Kconfig | 25 --------------
 arch/arm/cpu/mvebu-common/Kconfig     | 65 -----------------------------------
 drivers/ddr/Kconfig                   | 32 +++++++++++++++++
 drivers/gpio/Kconfig                  | 23 +++++++++++++
 drivers/misc/Kconfig                  | 16 +++++++++
 drivers/mmc/Kconfig                   |  9 +++++
 drivers/serial/Kconfig                |  9 +++++
 7 files changed, 89 insertions(+), 90 deletions(-)

diff --git a/arch/arm/cpu/armv8/armada3700/Kconfig b/arch/arm/cpu/armv8/armada3700/Kconfig
index ac11a5a..4302a5f 100644
--- a/arch/arm/cpu/armv8/armada3700/Kconfig
+++ b/arch/arm/cpu/armv8/armada3700/Kconfig
@@ -35,31 +35,6 @@ config PALLADIUM
 	  like Clock and PHY, this should
 	  not be enabled for real SoC.
 
-config MVEBU_SPL_A3700_GPIO
-	bool "GPIO configuration for A3700 in SPL"
-	depends on TARGET_ARMADA_3700
-	default y
-	help
-	  Choose this option to enable GPIO
-	  configuration for A3700 in SPL.
-	  This will be removed later, and be
-	  implemented by MPP/PIN-CTRL driver.
-
-config MVEBU_A3700_IO_ADDR_DEC
-	bool "Armada-3700 Address decoding support"
-	depends on !MVEBU_SPL_DDR_OVER_PCI_SUPPORT
-	default n
-	help
-	  Choose this option to add support
-	  for Marvell Armada-3700 address decoding driver
-
-config MVEBU_A3700_MISC_INIT
-	bool "Armada-3700 miscellaneous initialization procedures support"
-	default n
-	help
-	  Choose this option to add support
-	  for Marvell Armada-3700 miscellaneous init procedures
-
 config MVEBU_A3700_CLOCK
 	bool "Armada-3700  clock support"
 	default n
diff --git a/arch/arm/cpu/mvebu-common/Kconfig b/arch/arm/cpu/mvebu-common/Kconfig
index 95bcd6f..2f52fa1 100644
--- a/arch/arm/cpu/mvebu-common/Kconfig
+++ b/arch/arm/cpu/mvebu-common/Kconfig
@@ -31,71 +31,6 @@ config MVEBU_SPL_DIFFRENT_BASE_ADDR
 	  Choose this option to enable setting of a different
 	  registers base address at the SPL stage.
 
-menu "IO support"
-
-config MVEBU_MPP_BUS
-	bool "MPP bus support"
-	depends on (DEVEL_BOARD && MVEBU_PINCTL)
-	default n
-	help
-	  Choose this option to add support
-	  for the mpp-bus driver. This driver
-	  is used for enabling or disabling mpp
-	  buses at runtime
-
-config MVEBU_UART_ARLP
-	bool "UART support for ArmadaLP"
-	default n
-	help
-	  Choose this option to add support
-	  for UART driver on ArmadaLP, which
-	  is not using COM1 NS16550 driver.
-	  And base address is configured in FDT file.
-
-config MVEBU_SPL_DDR_OVER_PCI_SUPPORT
-	bool "DDR over PCIE support in SPL"
-	depends on SPL
-	default n
-	select MVEBU_PCI
-	help
-	  To add this option need to edit the defconfig and add the following line:
-	  +S:CONFIG_MVEBU_SPL_DDR_OVER_PCI_SUPPORT=y
-	  Choose this option to add support
-	  for DDR over PCIE support
-
-config MVEBU_SPL_DDR_OVER_PCI_SWITCH
-	bool "DDR over PCIE switch support in SPL"
-	depends on MVEBU_SPL_DDR_OVER_PCI_SUPPORT
-	default n
-	help
-	  Support DDR over PCIe switch, this should be selected in case the DDR
-	  PCIe addon card (PCAC) is located over a pcie switch (The first port of the
-	  switch).
-	  To add this option need to edit the defconfig and add the following line:
-	  +S:CONFIG_MVEBU_SPL_DDR_OVER_PCI_SWITCH=y
-
-config DDR_OVER_PCI_SIZE
-	int "size of DDR over PCI"
-	depends on MVEBU_SPL_DDR_OVER_PCI_SUPPORT
-	default  536870912
-	help
-	  Defines the size of DRAM over
-	  on the PCI endpoint. Relvant
-	  only when DDR over PCI is
-	  used.
-
-config XENON_MMC
-	bool "XENON MMC support"
-	default n
-	select CMD_MMC
-	help
-	  Choose this option to add support
-	  for Marvell XENON MMC driver, using Marvell xenon_mmc.c
-	  driver.
-	  base address is configured under XENON_MMC_BASE
-
-endmenu
-
 choice
 	prompt "Flash for image"
 	default MVEBU_SPI_BOOT
diff --git a/drivers/ddr/Kconfig b/drivers/ddr/Kconfig
index 3b55337..0247a0e 100644
--- a/drivers/ddr/Kconfig
+++ b/drivers/ddr/Kconfig
@@ -63,4 +63,36 @@ config MVEBU_DLL_TUNE_SUPPORT
 
 
 endchoice
+
+config MVEBU_SPL_DDR_OVER_PCI_SUPPORT
+	bool "DDR over PCIE support in SPL"
+	depends on SPL
+	default n
+	select MVEBU_PCI
+	help
+	  To add this option need to edit the defconfig and add the following line:
+	  +S:CONFIG_MVEBU_SPL_DDR_OVER_PCI_SUPPORT=y
+	  Choose this option to add support
+	  for DDR over PCIE support
+
+config MVEBU_SPL_DDR_OVER_PCI_SWITCH
+	bool "DDR over PCIE switch support in SPL"
+	depends on MVEBU_SPL_DDR_OVER_PCI_SUPPORT
+	default n
+	help
+	  Support DDR over PCIe switch, this should be selected in case the DDR
+	  PCIe addon card (PCAC) is located over a pcie switch (The first port of the
+	  switch).
+	  To add this option need to edit the defconfig and add the following line:
+	  +S:CONFIG_MVEBU_SPL_DDR_OVER_PCI_SWITCH=y
+
+config DDR_OVER_PCI_SIZE
+	int "size of DDR over PCI"
+	depends on MVEBU_SPL_DDR_OVER_PCI_SUPPORT
+	default  536870912
+	help
+	  Defines the size of DRAM over
+	  on the PCI endpoint. Relvant
+	  only when DDR over PCI is
+	  used.
 endmenu
diff --git a/drivers/gpio/Kconfig b/drivers/gpio/Kconfig
index a96fda3..2e3a273 100644
--- a/drivers/gpio/Kconfig
+++ b/drivers/gpio/Kconfig
@@ -1,3 +1,5 @@
+menu "GPIO Support"
+
 config DM_GPIO
 	bool "Enable Driver Model for GPIO drivers"
 	depends on DM
@@ -13,3 +15,24 @@ config MVEBU_PINCTL
 	  for pinctl driver. This drive implements
 	  the lowlevel control that allows changing
 	  the function of a multi purpose pin
+
+config MVEBU_MPP_BUS
+	bool "MPP bus support"
+	depends on (DEVEL_BOARD && MVEBU_PINCTL)
+	default n
+	help
+	  Choose this option to add support
+	  for the mpp-bus driver. This driver
+	  is used for enabling or disabling mpp
+	  buses at runtime
+
+config MVEBU_SPL_A3700_GPIO
+	bool "GPIO configuration for A3700 in SPL"
+	depends on TARGET_ARMADA_3700
+	default y
+	help
+	  Choose this option to enable GPIO
+	  configuration for A3700 in SPL.
+	  This will be removed later, and be
+	  implemented by MPP/PIN-CTRL driver.
+endmenu
diff --git a/drivers/misc/Kconfig b/drivers/misc/Kconfig
index df8abdb..f0f9824 100644
--- a/drivers/misc/Kconfig
+++ b/drivers/misc/Kconfig
@@ -58,6 +58,14 @@ config MVEBU_SPL_CHIP_SAR_DUMP
 	  Choose this option to dump sample at
 	  reset registers
 
+config MVEBU_A3700_MISC_INIT
+	bool "Armada-3700 miscellaneous initialization procedures support"
+	depends on TARGET_ARMADA_3700
+	default n
+	help
+	  Choose this option to add support
+	  for Marvell Armada-3700 miscellaneous init procedures
+
 endmenu
 
 menu "Address Decoding"
@@ -103,4 +111,12 @@ config MVEBU_MBUS
 	  This driver initialize the MBUS windows
 	  only.
 
+config MVEBU_A3700_IO_ADDR_DEC
+	bool "Armada-3700 Address decoding support"
+	depends on !MVEBU_SPL_DDR_OVER_PCI_SUPPORT && TARGET_ARMADA_3700
+	default n
+	help
+	  Choose this option to add support
+	  for Marvell Armada-3700 address decoding driver
+
 endmenu
diff --git a/drivers/mmc/Kconfig b/drivers/mmc/Kconfig
index e69de29..2761c20 100644
--- a/drivers/mmc/Kconfig
+++ b/drivers/mmc/Kconfig
@@ -0,0 +1,9 @@
+config XENON_MMC
+	bool "XENON MMC support"
+	default n
+	select CMD_MMC
+	help
+	  Choose this option to add support
+	  for Marvell XENON MMC driver, using Marvell xenon_mmc.c
+	  driver.
+	  base address is configured under XENON_MMC_BASE
diff --git a/drivers/serial/Kconfig b/drivers/serial/Kconfig
index 3f6901b..3bbf117 100644
--- a/drivers/serial/Kconfig
+++ b/drivers/serial/Kconfig
@@ -18,3 +18,12 @@ config UNIPHIER_SERIAL
 	depends on ARCH_UNIPHIER && DM_SERIAL
 	help
 	  Support for the on-chip UARTs on the Panasonic UniPhier platform.
+
+config MVEBU_UART_ARLP
+	bool "UART support for ArmadaLP"
+	default n
+	help
+	  Choose this option to add support
+	  for UART driver on ArmadaLP, which
+	  is not using COM1 NS16550 driver.
+	  And base address is configured in FDT file.
-- 
1.9.1

